### PR TITLE
fix(devtools): update menu list css

### PR DESCRIPTION
**Pull request by jyash97**
## Summary
In devtools when we select a deeply nested component, and try to select parent component using the dropdown. Text overlaps with the background text since the menu list is rendered outside the dom so CSS var is not applied properly.

Check below recording for the issue:

MenuListIssue

## How did you test this change?
Tested using inline devtools testing example. Below is the screen recording for testing the CSS:
MenuListThemeFix

### CODE REVIEW

**Comment by jyash97** [comment]
Hey @bvaughn  👋

Can you take a look at this changes, whenever you get time ?

File: packages/react-devtools-shared/src/devtools/views/Components/OwnersStack.css [hunk 1]
```diff
@@ -73,27 +73,32 @@
   background: var(--color-button-background-focus);
 }
 
-.Modal[data-reach-menu-list] {
+.ModalDark[data-reach-menu-list], .ModalLight[data-reach-menu-list] {
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/Components/OwnersStack.css:76]: Move color-specific stuff into a separate rule? Will also allow us to more easily compare light vs dark.

**Review by bvaughn — COMMENTED** [review]
[CI is failing with a lint error](https://app.circleci.com/pipelines/github/facebook/react/20397/workflows/8da23d35-a075-488e-bd6a-85f433f4366f/jobs/400137). Please run `yarn prettier` to fix.

File: packages/react-devtools-shared/src/devtools/views/Components/OwnersStack.css (hunk 1 above)
Comment (jyash97) [reply packages/react-devtools-shared/src/devtools/views/Components/OwnersStack.css:76]: Yes totally makes sense 🙌

**Comment by jyash97** [comment]
Many CI check seems to be failing now. Any idea?

Looks unrelated to the changes I did in latest commit

**Comment by bvaughn** [comment]
Can you try rebasing on `main` ?

```sh
git rebase main
```

**Review by bvaughn — COMMENTED** [review]
This is looking a lot better!

Now the only thing that's broken is the hover, I think. Probably need to do the same thing for the `MenuItem` CSS class as we did for the `MenuList` class.

**Comment by jyash97** [comment]
> Now the only thing that's broken is the hover, I think

Yes its left, should we add another variable in `root.css` for hover background ?

> It's really unfortunate how our CSS colors get forked between constants.js and root.css because of these global styles ☹️ I wonder if there's better solution here.


Yes I was unsure whether the solution I proposed was good since it adds the same values from `constants.js` to `root.css`. I have shared my thoughts here on the same: https://github.com/facebook/react/issues/22678#issuecomment-958141047

**Comment by bvaughn** [comment]
Love the suggestion on #22678.

**Comment by jyash97** [comment]
Glad it helped 😄   

Do I need to close this PR ? If not, can you help me with this:

> Should we add another variable in root.css for hover background ?

**Comment by bvaughn** [comment]
> Should we add another variable in root.css for hover background ?

For the "fix" to work fully with this PR, yes. We would need to do this as well. Or we could land the other fix first and then this would not be necessary.

**Comment by jyash97** [comment]
Closing as it will be fixed in #22716

**Comment by bvaughn** [comment]
Love it

**Pull request closed without merging**